haystack/document_stores/opensearch.py

```diff
@@ -119,8 +121,10 @@ def __init__(
                                     overwrite: Update any existing documents with the same ID when adding documents.
                                     fail: an error is raised if the document ID of the document being added already
                                     exists.
-        :param index_type: The type of index to be created. Choose from 'flat' and 'hnsw'.
-                           As OpenSearch currently does not support all similarity functions (e.g. dot_product) in exact vector similarity calculations,
+        :param index_type: The type of index you want to create. Choose from 'flat', 'hnsw', 'ivf', or 'ivf_pq'.
+                           'ivf_pq' is an IVF index optimized for memory through product quantization.
+                           ('ivf' and 'ivf_pq' are only available with 'faiss' as knn_engine.)
+                           As OpenSearch currently does not support all similarity functions (for example, dot_product) in exact vector similarity calculations,
```

Comment:
I must have forgotten this when making the score_script PR. Let's replace this and the following 3 lines by:
As OpenSearch does not require a special index for exact vector similarity calculations, we use a hnsw index optimized for memory footprint if index_type='flat'. Note that OpenSearchDocumentStore only executes exact vector calculations when knn_engine='score_script'. For the other knn_engines we execute hnsw if index_type='flat', as this usually achieves the best balance between latency and accuracy.